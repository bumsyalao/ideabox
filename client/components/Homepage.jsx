import React from 'react';
import { Link } from 'react-router-dom';
import ideaboxlogo from '../../client/images/idea logo.png';

/**
 * @function Homepage
 * @return {void}
 */
const Homepage = () => (
  <div>
    { $('.tooltipped').tooltip('remove') }
    <div className="section no-pad-bot bkg-style" id="index-banner">
      <div className="homepage">
        <br />
        <div className="center">
          <img src={ideaboxlogo} />
        </div>
        <div className="row center">
          <h3 className="header col s12 light">Think inside the box!</h3>
        </div>
        <div className="row center">
          <Link
            to="/auth"
            id="download-button"
            className="btn-large waves-effect waves-light amber black-text"
          >
            Get Started
          </Link>{' '}
        </div>
        <br />
      </div>
    </div>
  </div>
);

export default Homepage;
