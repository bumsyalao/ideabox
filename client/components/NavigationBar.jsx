import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router-dom';
import ideaboxlogo from '../../client/images/idea logo.png';
import { logoutRequest } from '../actions/userAction';

/**
 * NavigationBar component
 *
 * @class NavigationBar
 * @extends {Component}
 */
class NavigationBar extends Component {
  /**
   * Creates an instance of NavigationBar.
   * Binds clsss methods
   * @param {object} props
   * @return {void}
   * @memberOf Dashboard
   */
  constructor(props) {
    super(props);
    this.logout = this.logout.bind(this);
  }

  /**
   * Logout Function
   * Prevent default action of event
   * @param {object} event
   * @return {void}
   *
   * @memberOf NavigationBar
   */
  logout(event) {
    event.preventDefault();
    this.props.logoutRequest();
    this.props.history.push('/');
    Materialize.toast('You have logged out successfully', 3000, 'green');
  }

  /**
   *
   *
   * @returns {void}
   *
   * @memberOf NavigationBar
   */
  render() {
    $(document).ready(() => {
      $('.button-collapse').sideNav();
      $('.tooltipped').tooltip({ delay: 50 });
    });
    return (
      <div className="nav-style nav">
        <ul id="slide-out" className="side-nav">
          <li>
            <div className="user-view">
              <div className="background">
                <img src="images/office.jpg" />
              </div>
              <img className="img-logo" src={ideaboxlogo} />
              <a href="#!name">
                <span className="black-text name">{this.props.access.username}</span>
              </a>
              <a href="#!email">
                <span className="black-text email">{this.props.access.email}</span>
              </a>
            </div>
          </li>
          <li>
            <Link to="/dashboard/user-update" href="#!">
              <i className="material-icons">person_outline</i>Edit Profile
            </Link>
          </li>
          <li>
            <div className="divider" />
          </li>
          <li>
            <Link to="/dashboard/my-ideas" className="waves-effect" href="#!">
              <i className="material-icons">lightbulb_outline</i>
              My Ideas
            </Link>
          </li>
        </ul>
        <a
          href="#"
          className="button-collapse tooltipped"
          data-position="bottom"
          data-delay="50"
          data-tooltip="Menu"
          data-activates="slide-out"
        >
          <i className="material-icons">menu</i>
        </a>
        <div className="nav-items">
          <Link
            to="/dashboard/idea-list"
            href="#"
            className="tooltipped"
            data-position="bottom"
            data-delay="50"
            data-tooltip="Home"
          >
            <i className="material-icons">home</i>
          </Link>
          <Link
            to="/dashboard/create-idea"
            href="#"
            className="tooltipped"
            data-position="bottom"
            data-delay="50"
            data-tooltip="Create Idea"
          >
            <i className="material-icons">add_circle_outline</i>
          </Link>
          <a
            href="#"
            onClick={this.logout}
            className="tooltipped"
            data-position="bottom"
            data-delay="50"
            data-tooltip="Logout"
          >
            <i className="material-icons">power_settings_new</i>
          </a>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  access: state.access.user
});

export default connect(mapStateToProps, { logoutRequest })(withRouter(NavigationBar));

