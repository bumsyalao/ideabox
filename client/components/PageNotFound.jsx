import React from 'react';

const PageNotFound = () => (
  <div className="bkg-image">
    <h4 className="white-text"> Error 404: Page Not Found </h4>
  </div>

);

export default PageNotFound;
