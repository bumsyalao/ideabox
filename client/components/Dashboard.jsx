import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Route, Switch, withRouter } from 'react-router-dom';
import NavigationBar from '../components/NavigationBar';
import CreateIdea from '../components/Ideas/CreateIdea';
import IdeaList from '../components/Ideas/IdeaList';
import UserIdeaList from '../components/Ideas/UserIdeaList';
import EditIdea from '../components/Ideas/EditIdea';
import EditProfile from '../components/User/EditProfile';
import ViewIdea from '../components/Ideas/ViewIdea';
import PageNotFound from '../components/PageNotFound';

/**
 * Dashboard Component
 *
 * @class Dashboard
 * @extends {Component}
 */
class Dashboard extends Component {
  /**
   *
   * Check if user is authenticated
   * Toast access message
   * Intialize materialize collapsible
   * @return {void}
   * @memberOf Dashboard
   */
  componentDidMount() {
    if (!this.props.access.isAuthenticated) {
      Materialize.toast('Please Login or Register', 5000, 'red');
      this.props.history.push('/auth');
    }
  }
  /**
   *
   * Renders Dashboard Component
   * @returns {void}
   *
   * @memberOf Dashboard
   */
  render() {
    return (
      <div className="dashboard">
        <NavigationBar />
        <Switch>
          <Route
            path={`${this.props.match.url}/create-idea`}
            component={CreateIdea}
          />
          <Route
            path={`${this.props.match.url}/idea-list`}
            component={IdeaList}
          />
          <Route
            path={`${this.props.match.url}/my-ideas`}
            component={UserIdeaList}
          />
          <Route
            path={`${this.props.match.url}/edit/:ideaId`}
            component={EditIdea}
          />
          <Route
            path={`${this.props.match.url}/user-update`}
            component={EditProfile}
          />
          <Route
            path={`${this.props.match.url}/view/:ideaId`}
            component={ViewIdea}
          />
          <Route component={PageNotFound} />
        </Switch>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  access: state.access
});
export default connect(mapStateToProps, null)(withRouter(Dashboard));
