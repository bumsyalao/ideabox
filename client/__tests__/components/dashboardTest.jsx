/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import adapter from 'enzyme-adapter-react-16';
import { Dashboard } from '../../components/Dashboard';

configure({ adapter: new adapter() });

describe('Hompeage component', () => {
  const spy = jest.fn();
  global.Materialize = { toast: jest.fn() };
  beforeEach(() => {
    global.Materialize = { toast: jest.fn() };
  });
  afterEach(() => {
    spy.mockReset();
  });
  const props = {
    match: {
      path: '',
      url: ''
    },
    history: {
      push: spy
    },
    access: {
      isAuthenticated: false
    },
  };
  const component = shallow(
    <Dashboard {...props} />
  );

  it('should match the Dashboard component snapshot', () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
