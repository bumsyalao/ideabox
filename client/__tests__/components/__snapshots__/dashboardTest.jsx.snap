// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hompeage component should match the Dashboard component snapshot 1`] = `
<div
  className="dashboard"
>
  <Connect(withRouter(NavigationBar)) />
  <Switch>
    <Route
      component={[Function]}
      path="/create-idea"
    />
    <Route
      component={[Function]}
      path="/idea-list"
    />
    <Route
      component={[Function]}
      path="/my-ideas"
    />
    <Route
      component={[Function]}
      path="/edit/:ideaId"
    />
    <Route
      component={[Function]}
      path="/user-update"
    />
    <Route
      component={[Function]}
      path="/view/:ideaId"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
